src/infrastructure/transaction/CreateTransactionFromDTO.ts

```diff
@@ -349,20 +372,29 @@ const extractTransactionVersion = (version: number): number => {
  * @param recipient {string} Encoded hexadecimal recipient notation
  * @return {Address | NamespaceId}
  */
-const extractRecipient = (recipient: string): Address | NamespaceId => {
-    // If bit 0 of byte 0 is not set (like in 0x90), then it is a regular address.
-    // Else (e.g. 0x91) it represents a namespace id which starts at byte 1.
-    const bit0 = convert.hexToUint8(recipient.substr(1, 2))[0];
+const extractRecipient = (recipient: any): Address | NamespaceId => {
+    if (typeof recipient === 'string') {
+        // If bit 0 of byte 0 is not set (like in 0x90), then it is a regular address.
+        // Else (e.g. 0x91) it represents a namespace id which starts at byte 1.
+        const bit0 = convert.hexToUint8(recipient.substr(1, 2))[0];
 
-    if ((bit0 & 16) === 16) {
-        // namespaceId encoded hexadecimal notation provided
-        // only 8 bytes are relevant to resolve the NamespaceId
-        const relevantPart = recipient.substr(2, 16);
-        return NamespaceId.createFromEncoded(relevantPart);
-    }
+        if ((bit0 & 16) === 16) {
+            // namespaceId encoded hexadecimal notation provided
+            // only 8 bytes are relevant to resolve the NamespaceId
+            const relevantPart = recipient.substr(2, 16);
+            return NamespaceId.createFromEncoded(relevantPart);
+        }
 
-    // read address from encoded hexadecimal notation
-    return Address.createFromEncoded(recipient);
+        // read address from encoded hexadecimal notation
+        return Address.createFromEncoded(recipient);
+    } else if (typeof recipient === 'object') { // Is JSON object
+        if (recipient.hasOwnProperty('address')) {
+            return Address.createFromRawAddress(recipient.address);
+        }
+
+        return new NamespaceId(recipient.id);
```

Comment:
make sure to check `recipient.id` is available as well no ? (add it in `else if ..`).

There is a throw afterwards but in case `recipient.id` is not present, it will not throw.